docs/template_plugin/tests/functional/op_reference/concat.cpp

```diff
@@ -0,0 +1,1436 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/subtract.hpp"
+#include "base_reference_test.hpp"
+
+using namespace reference_tests;
+using namespace ov;
+
+namespace {
+struct ConcatParams {
+    ConcatParams(
+        const PartialShape& dynamicShape,
+        const Tensor& A, const Tensor& B, const Tensor& C, const int32_t axis,
+        const Tensor& expected, const std::string& testcaseName = "") :
+        dynamicShape(dynamicShape), A(A), B(B), C(C), axis(axis), expected(expected), testcaseName(testcaseName) {}
+
+    PartialShape dynamicShape;
+    Tensor A;
+    Tensor B;
+    Tensor C;
+    int32_t axis;
+    Tensor expected;
+    std::string testcaseName;
+};
+
+class ReferenceConcatTest : public testing::TestWithParam<ConcatParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        inputData = {params.A.data, params.B.data, params.C.data};
+        refOutData = {params.expected.data};
+    }
+
+    static std::string getTestCaseName(const testing::TestParamInfo<ConcatParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "dynShape=" << param.dynamicShape;
+        result << "_aType=" << param.A.type;
+        result << "_aShape=" << param.A.shape;
+        result << "_bType=" << param.B.type;
+        result << "_bShape=" << param.B.shape;
+        result << "_cType=" << param.C.type;
+        result << "_cShape=" << param.C.shape;
+        result << "_axis=" << param.axis;
+        result << "_eType=" << param.expected.type;
+        if (param.testcaseName != "") {
+            result << "_eShape=" << param.expected.shape;
+            result << "_=" << param.testcaseName;
+        } else {
+            result << "_rShape=" << param.expected.shape;
+        }
+        return result.str();
+    }
+
+private:
+    static std::shared_ptr<Function> CreateFunction(const ConcatParams& params) {
+        std::shared_ptr<op::v0::Parameter> A, B, C;
+        if (params.dynamicShape.is_dynamic()) {
+            A = std::make_shared<op::v0::Parameter>(params.A.type, params.dynamicShape);
```

Comment:
Could we use `ov::opsetN::` istead of `op::vN`?

Reply:
Updated.